tests/unit/test_network_connections.py

```diff
@@ -3670,6 +3707,95 @@ def test_path_to_glib_bytes(self):
         self.assertEqual(result.get_data(), b"file:///my/test/path\x00")
 
 
+class TestValidatorMatch(unittest.TestCase):
+    def setUp(self):
+        self.test_profile = {
+            "name": "test",
+            "type": "ethernet",
+            "ip": {
+                "dhcp4": False,
+                "address": "192.168.245.7/24",
+            },
+            "match": None,
+        }
+        self.validator = network_lsr.argument_validator.ArgValidator_DictConnection()
+
+    def test_match_path_empty_list(self):
+        """
+        Test that 'match.path' setting can be defined as None, [], [""], [None] or ""
+        and such a setting will be normalized into []
+        """
+        for disabled_path in [None, [], [""], [None], ""]:
+            self.test_profile["match"] = {"path": disabled_path}
+            result = self.validator.validate(self.test_profile)
+            self.assertEqual(result["match"], {"path": []})
+
+    def test_match_path_valid_setting(self):
+        """
+        Test that values like ["&"] and ["pci-0000:00:03.0"] are valid values for
+        'match.path' setting
+        """
+        self.test_profile["match"] = {"path": ["&"]}
+        result1 = self.validator.validate(self.test_profile)
+        self.assertEqual(result1["match"], {"path": ["&"]})
+
+        self.test_profile["match"] = {"path": ["pci-0000:00:03.0"]}
+        result2 = self.validator.validate(self.test_profile)
+        self.assertEqual(result2["match"], {"path": ["pci-0000:00:03.0"]})
+
+    def test_match_path_invalid_connection_type(self):
+        """
+        Test that when 'match.path' setting is correctly defined but the connection
+        type is neither ethernet nor infiniband, a ValidationError raises.
+        """
+        self.test_profile["match"] = {"path": ["pci-0000:00:03.0"]}
+        result = self.validator.validate(self.test_profile)
+        self.assertEqual(result["match"], {"path": ["pci-0000:00:03.0"]})
+
+        self.test_profile["type"] = "dummy"
+        self.assertRaisesRegex(
+            ValidationError,
+            "'match.path' settings are only supported for type 'ethernet' or "
+            "'infiniband'",
+            self.validator.validate,
+            self.test_profile,
+        )
+
+    def test_interface_name_when_match_not_specified(self):
+        """
+        Test that when 'match' setting is not specified, interface name should be
+        profile name.
+        """
+        result = self.validator.validate(self.test_profile)
+        self.assertEqual(result["interface_name"], "test")
+
+    def test_interface_name_when_match_is_None(self):
+        """
+        Test that when 'match' setting is None, interface name should be profile name.
+        """
+        self.test_profile["match"] = None
+        result = self.validator.validate(self.test_profile)
+        self.assertEqual(result["interface_name"], "test")
```

Comment:
As far as I understand, match would be normalized into "{}" when it is None, could you please add an assertion for that, too?

Reply:
Assertion was added. Not sure if it is correct way to do it or write an individual function. 
```yaml
    def test_interface_name_and_match_when_match_is_None(self):
        """
        Test that when 'match' setting is None, interface name should be profile name
        and 'match' setting will be normalized into {}.
        """
        self.test_profile["match"] = None
        result = self.validator.validate(self.test_profile)
        self.assertEqual(result["match"], {})
        self.assertEqual(result["interface_name"], "test")
```

Reply:
It is fine for me to do both assertions at the same time. Technically, it should probably be its own function to have better test names.